tests/components/geo_location/test_usgs_earthquakes_feed.py

```diff
@@ -0,0 +1,196 @@
+                    ATTR_UNIT_OF_MEASUREMENT: "km",
+                    ATTR_SOURCE: 'usgs_earthquakes_feed'}
+                assert round(abs(float(state.state)-20.5), 7) == 0
+
+                state = hass.states.get("geo_location.title_3")
+                assert state is not None
+                assert state.name == "Title 3"
+                assert state.attributes == {
+                    ATTR_EXTERNAL_ID: "3456", ATTR_LATITUDE: -31.2,
+                    ATTR_LONGITUDE: 150.2, ATTR_FRIENDLY_NAME: "Title 3",
+                    ATTR_UNIT_OF_MEASUREMENT: "km",
+                    ATTR_SOURCE: 'usgs_earthquakes_feed'}
+                assert round(abs(float(state.state)-25.5), 7) == 0
+
+                # Simulate an update - one existing, one new entry,
+                # one outdated entry
+                mock_feed.return_value.update.return_value = 'OK', [
+                    mock_entry_1, mock_entry_4, mock_entry_3]
+                async_fire_time_changed(hass, utcnow + SCAN_INTERVAL)
+                await hass.async_block_till_done()
+
+                all_states = hass.states.async_all()
+                assert len(all_states) == 3
+
+                # Simulate an update - empty data, but successful update,
+                # so no changes to entities.
+                mock_feed.return_value.update.return_value = 'OK_NO_DATA', None
+                async_fire_time_changed(hass, utcnow + 2 * SCAN_INTERVAL)
+                await hass.async_block_till_done()
+
+                all_states = hass.states.async_all()
+                assert len(all_states) == 3
+
+                # Simulate an update - empty data, removes all entities
+                mock_feed.return_value.update.return_value = 'ERROR', None
+                async_fire_time_changed(hass, utcnow + 3 * SCAN_INTERVAL)
+                await hass.async_block_till_done()
+
+                all_states = hass.states.async_all()
+                assert len(all_states) == 0
+
+
+async def test_setup_with_custom_location(hass):
+    """Test the setup with a custom location."""
+    # Set up some mock feed entries for this test.
+    mock_entry_1 = _generate_mock_feed_entry(
+        '1234', 'Title 1', 20.5, (-31.1, 150.1))
+
+    with patch('geojson_client.usgs_earthquake_hazards_program_feed.'
+               'UsgsEarthquakeHazardsProgramFeed') as mock_feed:
+        mock_feed.return_value.update.return_value = 'OK', [mock_entry_1]
+
+        with assert_setup_component(1, geo_location.DOMAIN):
+            assert await async_setup_component(
+                hass, geo_location.DOMAIN, CONFIG_WITH_CUSTOM_LOCATION)
+
+            # Artificially trigger update.
+            hass.bus.async_fire(EVENT_HOMEASSISTANT_START)
+            # Collect events.
+            await hass.async_block_till_done()
+
+            all_states = hass.states.async_all()
+            assert len(all_states) == 1
+
+            mock_feed.assert_called_with((15.1, 25.2),
```

Comment:
We prefer to not use the assertion methods of the mock cause a simple typo on the method can make the assertion pass even if it should fail. Do eg:
```py
from unittest.mock import call
...
assert mock_feed.call_args == call(...)
```

Reply:
OK